src/libraries/System.Diagnostics.Process/tests/ProcessWaitingTests.cs

```diff
@@ -54,6 +97,32 @@ public void MultipleProcesses_ParallelStartKillWait()
             Task.WaitAll(Enumerable.Range(0, Tasks).Select(_ => Task.Run(work)).ToArray());
         }
 
+        [Fact]
+        public async Task MultipleProcesses_ParallelStartKillWaitAsync()
+        {
+            const int Tasks = 4, ItersPerTask = 10;
+            Action work = async () =>
```

Comment:
```suggestion
            Func<Task> work = async () =>
```

This has to be a `Func<Task>`, right now you are creating async voids and the `Task.WhenAll` will return right away.

Reply:
Thanks for this! I've updated the code.